### PR TITLE
docs(schemas): Updates for switch to NPM SimpleSchema

### PR DESCRIPTION
**Pull request by aldeed**
- Updates for switch to NPM SimpleSchema
- Clarify proper ways to extend, override, replace, and reference schemas

Closes https://github.com/reactioncommerce/reaction/issues/3494
Connected to https://github.com/reactioncommerce/reaction/pull/3331

### CODE REVIEW

File: developer/architecture/schemas.md [hunk 1]
```diff
@@ -110,74 +109,135 @@ export const SchemaExample =  new SimpleSchema({
...
-        };
-      }
-    }
+    autoValue: createdAtAutoValue
   }
 });
 ```
 
-#### Reaction.shopIdAutoValue
+#### createdAtAutoValue
```
Comment (machikoyasuda) [inline developer/architecture/schemas.md]: I can't find this method defined or used in core anywhere: https://github.com/reactioncommerce/reaction/blob/v1.8.0/lib/collections/schemas/helpers.js

Did it get lost?!

**Comment by machikoyasuda** [comment]
@aldeed I removed `createdAtAutoValue` cos I couldn't find it anywhere. Or does it live somewhere else? Otherwise 👍

**Comment by aldeed** [comment]
@machikoyasuda It's added in my PR, which is not yet merged, but these docs are in preparation for when it's merged. I don't particularly care whether `createdAtAutoValue` is listed, though, as long as the one that was there, which is incomplete, isn't kept.

**Comment by machikoyasuda** [comment]
@aldeed Got it. Ah ~ I see it here in your PR: https://github.com/reactioncommerce/reaction/pull/3331/files#diff-38706249fadda8e9e297d0695c16a8d4

Marked this PR for `release: next` + Release 1.9 milestone and will add back `createdAtAutoValue`.

**Comment by zenweasel** [comment]
@aldeed This looks good. Are we going to have a seperate migration guide for existing users of the schema?

**Comment by aldeed** [comment]
FYI SimpleSchema is merged to release-1.9.0 branch now

**Pull request merged**